client/landing/stepper/declarative-flow/internals/steps-repository/readymade-template-generate-content/components/text-progress-bar/index.tsx

```diff
@@ -0,0 +1,47 @@
+import React from 'react';
+import './style.scss';
+
+type ProgressBarProps = {
+	target: number;
+	text: string;
+};
+
+const getProgressBarColor = ( progress: number ) => {
+	if ( progress <= 30 ) {
+		return 'red';
+	} else if ( progress <= 80 ) {
+		return 'yellow';
+	}
+	return 'green';
+};
+
+const getProgressBarMessage = ( progress: number ) => {
+	if ( progress <= 30 ) {
+		return 'Please provide more details to get better results.';
+	} else if ( progress <= 80 ) {
+		return 'Not bad! You are almost there. Keep going.';
+	}
+	return 'Great job! You are all set.';
+};
+
+const calculateProgress = ( text: string, target: number ) => {
+	return Math.min( ( text.length / target ) * 100, 100 );
```

Comment:
Counting characters this way often leads to wrong results in other languages like Japanese because one single character is equivalent to a few words:

<img width="409" alt="Screenshot 2024-08-02 at 13 32 37" src="https://github.com/user-attachments/assets/ce2daeac-ab10-43eb-a8d0-6206223fc66c">

I remember seeing a tool in Calypso which works better across different languages, but cannot remember right now where it is

Reply:
Turns out it was a native JS tool! https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/Segmenter

Reply:
Low priority, I will try to solve this in a follow-up.